File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        l = 0
        maxL = 0
        for i in range(1,len(prices)):
            if prices[l] < prices[i]:
                maxL = max(maxL,prices[i] - prices[l])
            else:
                l = i
        return maxL

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    let l = 0
    let r = 1
    let maxP = 0
    while(r < prices.length){
        if(prices[l] < prices[r]){
            let profit = prices[r] - prices[l]
            maxP = Math.max(maxP,profit)
        }
        else{
            l = r
        }
        r++
    }
    return maxP
};